libqtile/window.py

```diff
@@ -785,6 +810,7 @@ def __init__(self, window, qtile):
             win = qtile.windows_map.get(transient_for)
             if win is not None:
                 group = win._group
+                self.z = win.z + 1  # stay above parent
```

Comment:
Something similar to this is being done with `Floating.compute_client_position`, I think it would be a bit neater to remove this, but do the equivalent in that function in the floating layout, replacing what it currently does with the `above` variable, i.e. `Floating.compute_client_position` ought to set `client.z` as it does with X and Y. All the `above` stuff can be removed, (including from `Window.place`), as this whole PR supercedes it.

Reply:
Thanks for spotting this, seems a very good idea.
I'll read through the code for the floating layout. Kind of neglected that so far.

Reply:
So I have read through floating.py (I was surprised it isn't using `_ClientList`) and I followed your suggestions. This was indeed a very good thing to spot, things seem more natural now.

Reply:
I have yet to fully grasp what removing the `above`-flag from `Window.place` would mean. Will read a little more code for that.

Reply:
It seems *almost* possible to remove the `above`-flag, only the popup-class would still require it. Since these popups aren't managed by a group, I don't see a way right now to fully remove it.

Reply:
Maybe just remove the `above=True` bit from the Popup class, as it's likely that it is redundant anyhow.

Reply:
But then who is going to raise that window to the top?

Reply:
> I guess when `Popup.unhide` is called it gets manually sent to the top of the
> stack anyway, so it probably doesn't need the `above=True` anyway.

I'll give it a shot. Just a little afraid that I won't notice if anything breaks; I'm not sure if there is anything actually using the `Popup`-class.